Lib/test/test_inspect.py

```diff
@@ -674,6 +674,28 @@ def test_findsource_code_in_linecache(self):
         finally:
             del linecache.cache[co.co_filename]
 
+    def test_findsource_code_too_short(self):
+        lines = [
+            "",
+            "",
+            "",
+            "def foo(): pass",
+        ]
+        co = compile("\n".join(lines), "f.py", "exec")
+        linecache.cache[co.co_filename] = (
+            1, None, lines, co.co_filename
+        )
+        try:
+            _, lnum = inspect.findsource(co)
+            self.assertEqual(lnum, 3)
+            linecache.cache[co.co_filename] = (
+                1, None, [""] * lnum, co.co_filename
+            )
+            _, lnum = inspect.findsource(co)
+            self.assertEqual(lnum, 0)  # maybe OSError?
+        finally:
+            del linecache.cache[co.co_filename]
```

Comment:
I dislike this test because it makes test_inspect too dependent of the internal linecache cache format. Honestly, I'm not sure that it's worth it to have a test. I would be fine to only test the change manually.

**or** please rewrite the test without injecting data into linecache.cache, but create a file on disk instead. Basically, write a program which reproduce what you did manually.

Reply:
Happy to remove the test and test manually instead. Please note that we already have this dependency elsewhere in test_inspect (see: test_findsource_code_in_linecache for instance).